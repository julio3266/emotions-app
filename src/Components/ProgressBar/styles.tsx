import styled from 'styled-components/native';
import {Animated} from 'react-native';
import Icon from 'react-native-vector-icons/Entypo';
import Theme from '@theme/styles';

export const Progress = styled(Animated.View)``;

export const Row = styled.View`
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  bottom: 5%;
`;

export const PercentsContainer = styled.View`
  position: absolute;
  flex-direction: row;
  top: -80px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  justify-content: space-around;
  align-items: center;
`;

export const EmotionsContainer = styled.View`
  position: absolute;
  flex-direction: row;
  top: 40px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  justify-content: space-around;
  height: 30px;
  align-items: center;
`;

export const PercentTitle = styled.Text`
  font-size: ${(props: {selected: boolean}) =>
    props.selected ? '25px' : '20px'};
  bottom: ${(props: {selected: boolean}) => (props.selected ? '30px' : '18px')};
  color: ${(props: {selected: boolean}) =>
    props.selected ? Theme.colors.blueViolet : Theme.colors.gray};
  font-family: ${({theme}) => theme.fonts.robotoRegular};
`;

export const EmotionsData = styled.Text`
  color: ${({theme}) => theme.colors.white};
  font-family: ${({theme}) => theme.fonts.robotoRegular};
  font-size: ${(props: {selected: boolean}) => (props.selected ? '25px' : '0')};
  top: ${(props: {selected: boolean}) => (props.selected ? '30px' : '18px')};
  font-size: 15px;
  font-weight: 700;
`;

export const DragDropView = styled(Animated.View)`
  flex-direction: column;
  margin-right: auto;
  position: absolute;
  left: 3%;
`;

export const DragDropBtn = styled.TouchableOpacity`
  background-color: ${({theme}) => theme.colors.white};
  justify-content: center;
  align-items: center;
  border-radius: 50px;
  left: 35%
  width: 50px;
  top: 20%;
  height: 50px;
`;

export const IconBars = styled(Icon).attrs({
  name: 'menu',
  size: 30,
})`
  color: ${({theme}) => theme.colors.gray};
`;

export const WaveContainer = styled.View`
  position: absolute;
  bottom: 30px;
`;
