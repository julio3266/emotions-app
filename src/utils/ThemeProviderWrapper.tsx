import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
