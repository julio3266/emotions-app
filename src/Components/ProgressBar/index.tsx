import React from 'react';
import {
  GestureEvent,
  GestureHandlerRootView,
  PanGestureHandler,
  PanGestureHandlerEventPayload,
} from 'react-native-gesture-handler';

import Animated from 'react-native-reanimated';
import * as Styled from './styles';
import {IWave, Wave} from '@Components/Wave';
import {assignTestId} from '@utils/qualityAssurance';

interface IEmotions {
  emotionTitle: string;
  selected: boolean;
}
interface IPercents {
  percenValue: number;
  selected: boolean;
}

interface IProgressData {
  percents?: IPercents[];
  emotions?: IEmotions[];
}

export interface IProgressBar {
  testID?: string;
  progressData?: IProgressData[];
  gestureEvent?: (event: GestureEvent<PanGestureHandlerEventPayload>) => void;
  waveProps: IWave;
  animatedStyled?: any;
}

export const ProgressBar: React.FC<IProgressBar> = ({
  testID = 'ProgressBar',
  progressData,
  gestureEvent,
  animatedStyled,
  waveProps,
}) => {
  return (
    <GestureHandlerRootView {...assignTestId('View', testID)}>
      <Styled.PercentsContainer {...assignTestId('View', `${testID}_content`)}>
        {progressData?.[0]?.percents?.map(
          (percents: any, percentIndex: any) => {
            return (
              <Styled.PercentTitle
                {...assignTestId(
                  'Text',
                  `${testID}_percentTitle_${percentIndex}`,
                )}
                selected={percents.selected}>
                {percents.percenValue}
              </Styled.PercentTitle>
            );
          },
        )}
      </Styled.PercentsContainer>
      <Styled.Progress {...assignTestId('View', `${testID}_progressContainer`)}>
        <Styled.WaveContainer
          {...assignTestId('View', `${testID}_waveContainer`)}>
          <Wave {...assignTestId('View', `${testID}_wave`)} {...waveProps} />
        </Styled.WaveContainer>
        <Styled.Row {...assignTestId('View', `${testID}_row`)}>
          <PanGestureHandler
            {...assignTestId('View', `${testID}_panGestureHandler`)}
            onGestureEvent={gestureEvent}>
            <Animated.View
              {...assignTestId('View', `${testID}_animatedView`)}
              style={[animatedStyled]}>
              <Styled.DragDropBtn
                {...assignTestId(
                  'TouchableOpacity',
                  `${testID}_dragDropButton`,
                )}>
                <Styled.IconBars
                  {...assignTestId('Text', `${testID}_iconBars`)}
                />
              </Styled.DragDropBtn>
            </Animated.View>
          </PanGestureHandler>
        </Styled.Row>
        <Styled.EmotionsContainer
          {...assignTestId('View', `${testID}_emotionsContainer`)}>
          {progressData?.[0]?.emotions?.map(
            (percents: IEmotions, percentIndex: number) => {
              return (
                <Styled.EmotionsData
                  {...assignTestId(
                    'View',
                    `${testID}_emotionsData_${percentIndex}`,
                  )}
                  selected={percents.selected}>
                  {percents.emotionTitle}
                </Styled.EmotionsData>
              );
            },
          )}
        </Styled.EmotionsContainer>
      </Styled.Progress>
    </GestureHandlerRootView>
  );
};
