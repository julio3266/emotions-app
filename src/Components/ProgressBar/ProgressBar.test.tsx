import React from 'react';
import {ThemeProviderWrapper as wrapper} from '@utils/ThemeProviderWrapper';
import {mockTestID} from '@utils/qualityAssurance';
import {render} from '@testing-library/react-native';
import {ProgressBar, IProgressBar} from '@Components/ProgressBar';
import Theme from '@theme/styles';

describe('Components/ProgressBar', () => {
  let props: IProgressBar = {
    testID: 'ProgressBar',
    progressData: [
      {
        emotions: [{emotionTitle: 'Happy', selected: true}],
        percents: [{percenValue: 0, selected: true}],
      },
    ],
    animatedStyled: [],
    waveProps: {
      width: 100,
      height: 40,
      animationProps: {
        gestureWaveAnimated: jest.fn(),
        waveAnimationStyle: [],
      },
      color: Theme.colors.blueViolet,
    },
  };

  const {findByTestId, toJSON} = render(<ProgressBar {...props} />, {
    wrapper,
  });
  const sut_container = mockTestID('View', props.testID!).testID;
  const el_content = mockTestID('View', `${props.testID!}_content`).testID;
  const el_percentTitle = mockTestID(
    'Text',
    `${props.testID!}_percentTitle_0`,
  ).testID;
  const el_progressContainer = mockTestID(
    'Text',
    `${props.testID!}_progressContainer`,
  ).testID;

  const el_waveContainer = mockTestID(
    'View',
    `${props.testID!}_progressContainer`,
  ).testID;

  const el_wave = mockTestID('View', `${props.testID!}_wave`).testID;
  const el_row = mockTestID('View', `${props.testID!}_row`).testID;
  const el_panGestureHandler = mockTestID(
    'View',
    `${props.testID!}_panGestureHandler`,
  ).testID;
  const el_animatedView = mockTestID(
    'View',
    `${props.testID!}_animatedView`,
  ).testID;

  const el_dragDropButton = mockTestID(
    'TouchableOpacity',
    `${props.testID!}_dragDropButton`,
  ).testID;

  const el_iconBars = mockTestID('Text', `${props.testID!}_iconBars`).testID;
  const el_emotionsContainer = mockTestID(
    'View',
    `${props.testID!}_emotionsContainer`,
  ).testID;

  const el_emotionsData = mockTestID(
    'View',
    `${props.testID!}_emotionsData_0`,
  ).testID;

  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_content}`, () => {
    const sut = findByTestId(el_content);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_percentTitle}`, () => {
    const sut = findByTestId(el_percentTitle);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_progressContainer}`, () => {
    const sut = findByTestId(el_progressContainer);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_waveContainer}`, () => {
    const sut = findByTestId(el_waveContainer);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_row}`, () => {
    const sut = findByTestId(el_row);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_wave}`, () => {
    const sut = findByTestId(el_wave);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_panGestureHandler}`, () => {
    const sut = findByTestId(el_panGestureHandler);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_animatedView}`, () => {
    const sut = findByTestId(el_animatedView);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_dragDropButton}`, () => {
    const sut = findByTestId(el_dragDropButton);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_iconBars}`, () => {
    const sut = findByTestId(el_iconBars);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_emotionsContainer}`, () => {
    const sut = findByTestId(el_emotionsContainer);
    expect(sut).toBeTruthy();
  });

  it(`should render #${el_emotionsData}`, () => {
    const sut = findByTestId(el_emotionsData);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
