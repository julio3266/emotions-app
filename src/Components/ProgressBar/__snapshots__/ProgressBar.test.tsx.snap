// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ProgressBar should render a snapshot 1`] = `null`;
